.github/actions/run_with_e2e_account/action.yml

```diff
@@ -0,0 +1,75 @@
+name: run_with_e2e_account
+description: Runs commands with e2e test account
+inputs:
+  run:
+    description: Script to run
+    required: true
+  shell:
+    description: Shell
+    required: false
+  node_version:
+    description: node version used to configure environment with
+    required: false
+  e2e_test_accounts:
+    description: Serialized JSON array of strings with account numbers
+    required: true
+  aws_region:
+    description: AWS region. If not provided random will be selected
+    required: false
+  link_cli:
+    description: Whether should link Gen2 CLI globally
+    default: false
+  fresh_build:
+    description: Whether should build from scratch
+    default: false
+runs:
+  using: composite
+  steps:
+    - name: Setup Node.js
+      uses: ./.github/actions/setup_node
+      with:
+        node-version: ${{ inputs.node_version }}
+    - name: Restore Build Cache
+      if: inputs.fresh_build != 'true'
+      uses: ./.github/actions/restore_build_cache
+    - name: Build With Cache
+      if: inputs.fresh_build == 'true'
+      uses: ./.github/actions/build_with_cache
+    - name: Link CLI
+      if: inputs.link_cli == 'true'
+      shell: bash
+      run: cd packages/cli && npm link
+    - name: Select E2E test account
+      uses: ./.github/actions/select_e2e_account
+      id: selectE2EAccount
+      with:
+        e2e_test_accounts: ${{ inputs.e2e_test_accounts }}
+    - name: Select region
+      id: selectE2ERegion
+      shell: bash
+      run: |
+        if [ -z "${{ inputs.aws_region }}" ]; then
+          regions=("us-west-2" "us-east-1" "ca-central-1" "eu-central-1")
+          rand=$[$RANDOM % ${#regions[@]}]
+          selected_aws_region=${regions[$rand]}
+        else
+          selected_aws_region="${{ inputs.aws_region }}"
+        fi
```

Comment:
I chose bash for this, because it doesn't involve any JSON parsing like we do in case of account selection.
And is rather simple at this point.